proposals/PickAPI/README.md

```diff
@@ -0,0 +1,192 @@
+
+## Proposal
+To describe high level, customized primitive pick behavior, provide a `PickAPI` schema in `UsdUI`. This schema describes two key functions [visibility](#visibility) and [retargeting](#retargeting).
+
+This proposal distinguishes between pick specialization and [application selection modes](#relationship-to-application-selection-modes). Selection modes are user interface options that apply to the entire scene and cannot be practically encoded on every prim in the scene.
+
+The `PickAPI` should only be applied to `Imageable` prims (ie. `UsdGeom` and `UsdVol` but not `UsdShade`).
+
+### Visibility
+```
+uniform token pick:visibility = "inherited" (allowedTokens = ["inherited", "invisible"])
+```
+`pick:visibility` controls whether or not the object should be considered visible with respect to any picking operation. Objects `invisible` to picking should be excluded from any picking calculations (and acceleration structures like BVHs or pick buffers). Prims should only be included in the pick buffer if they are imaged, taking into account `visibility`, `purpose`, and any other viewport render settings.
+
+Because picking is often accelerated through the rendering toolkit, it's important to describe picking on its terms.
+
+If the `PickAPI` is applied to multiple sites in a hierarchy, `pick:visibility` has pruning semantics similar to the `visibility` attribute. If a prim is invisible to picking, so are its descendants.
+
+To make an object _unpickable_ but still participate in calculations as a matte occluder, users must use [retargeting](#retargeting-for-pick-occluders).
+
+`pick:visibility` may not be animated.
+
+In this current proposal, sending geometry to a pick buffer that is not actively imaged is currently out of scope. One hypothetical use case would be interactive picking of volumetrics. A user might provide a mesh to be used as the pick target for a dense volumetric cloud. This proposal doesn't provide a path for including non-imageable geometry in picking because the semantics become somewhat confusing and complicates implementations leveraging rendering pipelines.
+
+### Retargeting
+
+```
+uniform token pick:retargeting = "none" (allowedTokens = ["none", "replace"])
+rel pick:targets
+```
+These attributes control whether or not to forward picking to alternative targets. The default behavior (`none`) is to not forward picking. However, if `replace` is selected, all descendant prim selections will be replaced with `pick:targets`.
+
+If the `PickAPI` is applied to multiple sites in a hierarchy, the retargeting behavior will be defined by the nearest applied ancestor. This is notably different than the pruning semantics of `pick:visibility`. As specified, retargeting could be set to `none` in a descendant, overriding any ancestral retargeting opinions.
```

Comment:
Given that retargeting is non-pruning, why not make the `pick:visibility` non-pruning, as well, to have the same semantics for all picking aspects?  While `imageable.visibility` _is_ pruning, `imageable.purpose` is not.  If we called it `pickable` instead of `visibility`, I don't think there would be any confusion.  I believe this would also provide more flexible in setting up mattes.

Reply:
This is a good question.  I considered at one point having a few modes for `pick:visibility`. One hypothetical use case was to provide a pick proxy for a hard to pick prim like a volume.  I thought this made things more confusing, as it required a mode that would override / ignore other visibility state as you wanted it to only render in the pick buffer (but only sometimes ignore the other visibility state).

One could argue that this "pick buffer only" rendering could or should be an extension to `purpose`.

I ultimately settled on strict pruning visibility as I suspected the simplest formulation would be the most successful path to introduce new behavior into rendering contexts.